### ISSUE TITLE
The problem of generating vocab when using transformer.

### ISSUE THREAD

**Issue by Elenore1997** [dmlc/dgl#516]
I saw a line of code in transformer/dataset/__init__.py, about create vocab
`
def _make_vocab(self, path, thres=2):
        word_dict = {}
        for mode in ['train', 'valid', 'test']:
`
I think the words in vocab are only obtained by training set.

**Comment by yzh119** [comment]
Well, it does not matter if we obtain vocabulary from both training and valid/test set.
If a word does not appear in training set, then its embedding would not be updated and it is not likely to be generated in inference phase. You would get similar results if you build vocabulary only from training set.
For cases where pre-trained embeddings are used, building vocabulary from both training and valid/test dataset is useful, [this issue](https://github.com/allenai/allennlp/issues/516) might be useful for you to understand this point.
However in real translation settings, people use subword level vocabulary and you should no longer be worried about this problem.

**Comment by Elenore1997** [comment]
Thanks for your reply! It makes me clear about this problem.
Sry to bother and I think this issue can be closed.
